### PR TITLE
[Addon] Feature templates enhancement

### PR DESCRIPTION
**Pull request by cicharka**
# Pull Request summary:
Feature Templates needed enhancements and fixes in order to use them in Ansible collections.

# Description of changes:
* enhance all models existing models with `_docs_description` field that will be used in Ansible documentation generation
* enhance all models fields with `description` Filed attribute for Ansible documentation generation
* remove all Enum classes and move them to Literals
* move common `LIterals` to `catalystwan/models/common.py`
* enhance tests for templates with all available models
* fix already running tests to include new models
* add example of Feature Templates in README and provide link to FT README page
* enhance cli_templates api to read cli configuration from file
* Add workflow for backup & restore of Device Templates that are based on Feature Templates and policies

## NOTE: THIS PR IS INTRODUCING BREAKING CHANGES FOR EXISTING CODE:
* due to migration from `Enum` to `Literal`

# Checklist:
- [x] Make sure to run pre-commit before committing changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

**Review by sbasan — APPROVED** [review]

File: catalystwan/api/templates/models/cisco_aaa_model.py [hunk 1]
```diff
@@ -49,36 +73,61 @@ class DomainStripping(str, Enum):
...
+        default=None,
+        json_schema_extra={"vmanage_key": "source-interface"},
+        description="The source interface for the TACACS+ group",
+    )
+    server: List[TacacsServer] = Field(default=[], description="The list of TACACS+ servers for the group")
 
 
 class CiscoAAAModel(FeatureTemplate):
     model_config = ConfigDict(arbitrary_types_allowed=True, populate_by_name=True)
+    docs_description: str = Field(exclude=True, description="Cisco AAA Feature Template configuration.")
```
Comment (sbasan) [inline catalystwan/api/templates/models/cisco_aaa_model.py:109]: I don't think that belongs to model. Use private field?
https://docs.pydantic.dev/latest/concepts/models/#private-model-attributes
```suggestion
    _docs_description: str = "Cisco AAA Feature Template configuration."
```

**Comment by cicharka** [comment]
Still WIP, need to add all models.

File: catalystwan/api/templates/models/cisco_vpn_interface_model.py [hunk 2]
```diff
@@ -22,443 +21,826 @@
 DEFAULT_IPV6_VRRP_TIMER = 1000
 
 
-class SecondaryIPv4Address(FeatureTemplateValidator):
-    address: Optional[ipaddress.IPv4Interface] = None
+Direction = Literal["in", "out"]
+NatChoice = Literal["Interface", "Pool", "Loopback"]
+StaticNatDirection = Literal["inside", "outside"]
+Proto = Literal["tcp", "udp"]
+CoreRegion = Literal["core", "core-shared"]
```
Comment (jpkrajewski) [inline catalystwan/api/templates/models/cisco_vpn_interface_model.py:28]: Please move this to common

File: catalystwan/api/templates/models/cisco_vpn_interface_model.py [hunk 3]
```diff
@@ -22,443 +21,826 @@
...
+    "custom2",
+    "custom3",
+    "private1",
+    "private2",
+    "private3",
+    "private4",
+    "private5",
+    "private6",
+]
+Carrier = Literal[
```
Comment (jpkrajewski) [inline catalystwan/api/templates/models/cisco_vpn_interface_model.py:56]: Move to common

File: catalystwan/api/templates/models/system_vsmart_model.py [hunk 4]
```diff
@@ -1,45 +1,189 @@
...
 
 from pydantic import ConfigDict, Field
 
-from catalystwan.api.templates.feature_template import FeatureTemplate
+from catalystwan.api.templates.feature_template import FeatureTemplate, FeatureTemplateValidator
 from catalystwan.utils.timezone import Timezone
 
+Toology = Literal["Hub and Spoke"]
+
+TlocColor = Literal[
```
Comment (jpkrajewski) [inline catalystwan/api/templates/models/system_vsmart_model.py:13]:
```suggestion
from catalystwan.models.common import TLOCColor
```

File: catalystwan/api/templates/models/cisco_bfd_model.py [hunk 5]
```diff
@@ -16,51 +15,72 @@
...
-    CUSTOM1 = "custom1"
-    CUSTOM2 = "custom2"
-    CUSTOM3 = "custom3"
-    PRIVATE1 = "private1"
-    PRIVATE2 = "private2"
-    PRIVATE3 = "private3"
-    PRIVATE4 = "private4"
-    PRIVATE5 = "private5"
-    PRIVATE6 = "private6"
+ColorType = Literal[
```
Comment (jpkrajewski) [inline catalystwan/api/templates/models/cisco_bfd_model.py:18]:
```suggestion
from catalystwan.models.common import TLOCColor
```

File: catalystwan/api/templates/models/cisco_vpn_interface_model.py [hunk 6]
```diff
@@ -22,443 +21,826 @@
...
-    address: Optional[ipaddress.IPv4Interface] = None
+Direction = Literal["in", "out"]
+NatChoice = Literal["Interface", "Pool", "Loopback"]
+StaticNatDirection = Literal["inside", "outside"]
+Proto = Literal["tcp", "udp"]
+CoreRegion = Literal["core", "core-shared"]
+SecondaryRegion = Literal["off", "secondary-only", "secondary-shared"]
+Encap = Literal["gre", "ipsec"]
+Mode = Literal["hub", "spoke"]
+Value = Literal[
```
Comment (jpkrajewski) [inline catalystwan/api/templates/models/cisco_vpn_interface_model.py:32]:
```suggestion
from catalystwan.models.common import TLOCColor
```

**Review by jpkrajewski — COMMENTED** [review]
Please move as much as possible literals to common

**Review by jpkrajewski — COMMENTED** [review]
can we make simple POST integration tests for those models?

**Comment by cicharka** [comment]
> can we make simple POST integration tests for those models?

Yes, we can reuse already created models from [catalystwan/tests/templates/models/__init__.py](https://github.com/cisco-open/cisco-catalyst-wan-sdk/pull/39/files#diff-f3baf615498d9a5f9a36a6443471718deb197791880addd81cd11dba1a4eb6d3) - and just send them to Manager and see if they work. But these dummy values have not been tested yet with Device Template attachment so it might require more tweaks if we need 'working' Manager after attaching them.

**Comment by jpkrajewski** [comment]
> > can we make simple POST integration tests for those models?
> 
> Yes, we can reuse already created models from [catalystwan/tests/templates/models/**init**.py](https://github.com/cisco-open/cisco-catalyst-wan-sdk/pull/39/files#diff-f3baf615498d9a5f9a36a6443471718deb197791880addd81cd11dba1a4eb6d3) - and just send them to Manager and see if they work. But these dummy values have not been tested yet with Device Template attachment so it might require more tweaks if we need 'working' Manager after attaching them.

im thinking something like

```
def test_create():
    # Arrange
    created_model = Model()
    # Act
    id = api.create(created_model)
    response_model = api.get(id)
    # Assert
    assert response_model = created_model
```

**Comment by cicharka** [comment]
> > > can we make simple POST integration tests for those models?
> > 
> > 
> > Yes, we can reuse already created models from [catalystwan/tests/templates/models/**init**.py](https://github.com/cisco-open/cisco-catalyst-wan-sdk/pull/39/files#diff-f3baf615498d9a5f9a36a6443471718deb197791880addd81cd11dba1a4eb6d3) - and just send them to Manager and see if they work. But these dummy values have not been tested yet with Device Template attachment so it might require more tweaks if we need 'working' Manager after attaching them.
> 
> im thinking something like
> 
> ```
> def test_create():
>     # Arrange
>     created_model = Model()
>     # Act
>     id = api.create(created_model)
>     response_model = api.get(id)
>     # Assert
>     assert response_model = created_model
> ```

Issue created -> https://github.com/cisco-open/cisco-catalyst-wan-sdk/issues/656
Can be done in next PR.

**Review by jpkrajewski — DISMISSED** [review]

**Review by jpkrajewski — DISMISSED** [review]

**Comment by cicharka** [comment]
Merged to `dev` branch with https://github.com/cisco-open/cisco-catalyst-wan-sdk/commit/e16aabedc1cffd0c3d199dce120768d810ea29d5

**Pull request closed without merging**